lib/gitlib.sh

```diff
@@ -103,17 +129,24 @@ gitlib::last_releases () {
   declare -Ag LAST_RELEASE
 
   logecho -n "Setting last releases by branch: "
-  for release in $($GHCURL $K8S_GITHUB_API/releases|\
+  for release in $($GHCURL $K8S_GITHUB_API_ROOT/$PROJECT/$PROJECT/releases |\
                    jq -r '.[] | select(.draft==false) | .tag_name'); do
     # Alpha releases only on master branch
+    # NOTE: The logic here is mostly Kubernetes specific for now
+    # As I understand it, Istio's branching and tagging strategy is in flux
+    # so don't try to guess what it is and will be here for now.
+    # As of 2017/10/12 last release on master will end up being the x.0 release
+    # and on branches will be the last x.y release. That may even be the desired
+    # result and of course this is only the 'feature' of guessing tag
+    # boundaries.  The user can always put specific ranges on the command-line.
     if [[ $release =~ -alpha ]]; then
       LAST_RELEASE[master]=${LAST_RELEASE[master]:-$release}
-    elif [[ $release =~ v([0-9]+\.[0-9]+)\.([0-9]+(-.+)?) ]]; then
+    elif [[ $release =~ (${VER_REGEX[x.y]}\.${VER_REGEX[digit]}(-.+)?) ]]; then
       # Latest vx.x.0 release goes on both master and release branch.
-      if [[ ${BASH_REMATCH[2]} == "0" ]]; then
+      if [[ ${BASH_REMATCH[4]} == "0" ]]; then
```

Comment:
it's really hard to tell what part of the regexp `${BASH_REMATCH[4]}` is referring to.

Reply:
You are right.  Cleaned up and reduced clutter there.